### ISSUE TITLE
`dialog-button-role` possible on `StandardButton`

### ISSUE THREAD

**Issue by Enyium** [slint-ui/slint#6556]
Consider [this SlintPad demo](https://slintpad.com/?snippet=import+%7B+StandardButton+%7D+from+%22std-widgets.slint%22%3B%0A%0Aexport+component+Dlg+inherits+Dialog+%7B%0A++++Text+%7B%0A++++++++text%3A+%22%5BDialog+content%5D%22%3B%0A++++%7D%0A%0A++++StandardButton+%7B%0A++++++++kind%3A+ok%3B%0A++++++++dialog-button-role%3A+reject%3B%0A++++%7D%0A%7D%0A&style=native):

```slint
import { StandardButton } from "std-widgets.slint";

export component Dlg inherits Dialog {
    Text {
        text: "[Dialog content]";
    }

    StandardButton {
        kind: ok;
        dialog-button-role: reject;
    }
}
```

It doesn't generate an error. This is contradictory and should probably be forbidden.

**Comment by ogoffart** [comment]
Thanks for the bug report.
You're right that this is contradictory, but i don't think it is worth to forbid it.
At best there could be a warning that `dialog-button-role` shouldn't be necessaey
